### PR TITLE
[BugFix] Fix sink commit function without metadata

### PR DESCRIPTION
**Pull request by letian-jiang**
Fixes #issue

img_v2_780cf0ba-0975-46b9-8d92-758d8f07075g

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/exec/pipeline/sink/hive_table_sink_operator.cpp [hunk 1]
```diff
@@ -199,18 +199,20 @@ void HiveTableSinkOperatorFactory::close(RuntimeState* state) {
...
-    hive_file_info.__set_record_count(writer->metadata()->num_rows());
-    hive_file_info.__set_file_size_in_bytes(writer->file_size());
-
-    TSinkCommitInfo commit_info;
-    commit_info.__set_hive_file_info(hive_file_info);
-
-    // update runtime state
-    state->add_sink_commit_info(commit_info);
-    state->update_num_rows_load_sink(hive_file_info.record_count);
+    if (writer->metadata()) {
```
Comment (imay) [inline be/src/exec/pipeline/sink/hive_table_sink_operator.cpp:202]: 1. please add comments to explain when metadata() will be null.
2. prefer `writer->metadata() != nullptr` that explicitly compares a pointer with the nullptr.

**Review by imay — COMMENTED** [review]
Please add UT or e2e test to cover this fix.

File: be/src/exec/pipeline/sink/hive_table_sink_operator.cpp (hunk 1 above)
Comment (letian-jiang) [reply be/src/exec/pipeline/sink/hive_table_sink_operator.cpp]: I refactored a little bit, there would case in daily-test to cover this fix.

File: be/src/formats/parquet/file_writer.cpp [hunk 2]
```diff
@@ -496,20 +496,20 @@ Status AsyncFileWriter::close(RuntimeState* state,
...
-        _file_metadata = _writer->metadata();
         auto st = _outstream->Close();
         if (!st.ok()) {
             LOG(WARNING) << "close output stream error: " << st.message();
             set_io_status(Status::IOError(fmt::format("{}: {}", "close output stream error", st.message())));
         }
 
-        if (cb != nullptr) {
+        // invoke callback if status is ok
+        if (cb != nullptr && get_io_status().ok()) {
```
Comment (imay) [inline be/src/formats/parquet/file_writer.cpp:512]: My question is if someday this callback function will be called even when get_io_status() is not ok, what will happen? Are you sure if the developer who will change it knows the dependence you added here?

Comment (letian-jiang) [reply be/src/formats/parquet/file_writer.cpp:512]: - The callback function collects stats of each file, and there is no need to invoke it if the io status is not ok. 
- I put `DCHECK(writer->metadata() != nullptr);` in the callback to state my intention.

Comment (imay) [reply be/src/formats/parquet/file_writer.cpp:512]: My suggestion is why not make callback function have return value other than void

Comment (letian-jiang) [reply be/src/formats/parquet/file_writer.cpp:512]: 👀 Let's merge this patch, and I will refactor this part this week.

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**